trunkfingerprint--1.2.0.sql

```diff
@@ -90,6 +89,13 @@ begin
                                                                 then '(select oprname from pg_operator where oid = foo.' || attname || ')'
                                  when attname like '%opc'       then '(select opcname from pg_opclass where oid = foo.' || attname || ')'
                                  when attname like '%tablespace'then '(select spcname from pg_tablespace where oid = foo.' || attname || ')'
+                                 when attname like '%partdefid' then '(select relname from pg_class where oid = foo.' || attname || ')'
+                                 when attname like '%pnnspid'   then '(select nspname from pg_namespace where oid = foo.' || attname || ')'
+                                 when attname like '%pnpubid'   then '(select pubname from pg_publication where oid = foo.' || attname || ')'
+                                 when attname like '%tgparentid'
+                                                                then '(select concat(ns.nspname, ''.'' ,tg.tgname)
```

Comment:
As far as I understand, trigger name is not guaranteed to be unique within a namespace, only within triggers of a given table. Also, concatenating like this is prone to ambiguity, as namespace or trigger name may contain a dot themselves. This is not mere pedantry, we do have objects with dots in their names, albeit not triggers: these are foreign tables in schemas like `centraldb`.

I'd suggest `select  (tg.tgrelid::regclass::text, tg.tgname)::text from ...`. Casting a tuple into text is always reversible, as PG cares about escaping etc. So is casting a table oid to text via regclass, it escapes and adds schema prefix automatically where needed.